samples/azure-quantum/entanglement-swap/ES-quantinuum-qsharp.ipynb

```diff
@@ -0,0 +1,1120 @@
+   "execution_count": 8,
+   "metadata": {},
+   "outputs": [
+    {
+     "data": {
+      "application/x-qsharp-data": "[]",
+      "text/html": [
+       "<ul></ul>"
+      ],
+      "text/plain": []
+     },
+     "execution_count": 8,
+     "metadata": {},
+     "output_type": "execute_result"
+    }
+   ],
+   "source": [
+    "open Microsoft.Quantum.Convert;"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 9,
+   "metadata": {},
+   "outputs": [
+    {
+     "data": {
+      "application/x-qsharp-data": "[\"TeleportFidelityHelper\",\"TestFidelity\"]",
+      "text/html": [
+       "<ul><li>TeleportFidelityHelper</li><li>TestFidelity</li></ul>"
+      ],
+      "text/plain": [
+       "TeleportFidelityHelper, TestFidelity"
+      ]
+     },
+     "execution_count": 9,
+     "metadata": {},
+     "output_type": "execute_result"
+    }
+   ],
+   "source": [
+    "operation TeleportFidelityHelper() : Result {\n",
+    "    use (alice, bob, payload) = (Qubit(), Qubit(), Qubit());\n",
+    "    Ry(1.727, payload);\n",
+    "    CreateBellState(alice, bob);\n",
+    "    PerformTeleportationOnBellState(alice, bob, payload);\n",
+    "    Adjoint Ry(1.727, bob);\n",
+    "    return M(bob);\n",
+    "}\n",
+    "\n",
+    "operation TestFidelity(n : Int) : Double {\n",
+    "    mutable expected = 0;\n",
+    "    for i in 1..n {\n",
+    "        if (TeleportFidelityHelper() == Zero) {\n",
+    "            set expected = expected + 1;\n",
+    "        }\n",
+    "    }\n",
+    "    return IntAsDouble(expected) / IntAsDouble(n);\n",
+    "}"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Let us now compare how the theoretical fidelity compares to the practical fidelity. For this we will first simulate our code and then proceed to run it on Quantinuum hardware through Azure Quantum."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "%simulate TestFidelity n=100"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "We see that in theory on a perfect machine we'd see perfect fidelity. Now let us try running our work on Azure Quantum.\n",
+    "For that we begin by connecting to our Azure workspace."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "%azure.connect \"\" location=\"\""
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "When running jobs on Azure Quantum, before going to real hardware it is recommended to run your code through a validator to make sure you don't waste precious resources on faulty code. `quantinuum.hqs-lt-s2-apival` is the appropriate target for this."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "%azure.target quantinuum.hqs-lt-s2-apival"
```

Comment:
This should be as early in the notebook as possible so as to help give accurate target diagnostics.

Reply:
We can't do that here as we create code that's plainly for the local simulator that would throw errors